key-service.js

```diff
@@ -23,11 +24,13 @@ function factory(
     options = options || {};
     var identityId;
     var url;
+    var params = $httpParamSerializer(options.params)
     if(options.identity) {
-      url = basePath + '?owner=' + encodeURIComponent(options.identity.id);
+      url = basePath +
+        '?owner=' + encodeURIComponent(options.identity.id) + '&' + params;
```

Comment:
Should the params part here and below be added conditionally only if params exist?  Otherwise you get trailing & and ?.